sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.Response.cs

```diff
@@ -2,69 +2,110 @@
-                    // Make sure we don't dispose the content if the stream was replaced
-                    _responseMessage.Content = null;
+                get => _pipelineResponse.ContentStream;
+                set => _pipelineResponse.ContentStream = value;
+            }
 
-                    _contentStream = value;
+            public override BinaryData Content
+            {
+                get
+                {
+                    ResetContentStreamPosition(_pipelineResponse);
+                    return _pipelineResponse.Content;
                 }
             }
 
-            public override string ClientRequestId { get; set; }
+            public override BinaryData BufferContent(CancellationToken cancellationToken = default)
+                => _pipelineResponse.BufferContent(cancellationToken);
+
+            public override async ValueTask<BinaryData> BufferContentAsync(CancellationToken cancellationToken = default)
+                => await _pipelineResponse.BufferContentAsync(cancellationToken).ConfigureAwait(false);
 
-            protected internal override bool TryGetHeader(string name, [NotNullWhen(true)] out string? value) => HttpClientTransport.TryGetHeader(_responseMessage.Headers, _responseContent, name, out value);
+            protected internal override bool ContainsHeader(string name)
+                => _pipelineResponse.Headers.TryGetValue(name, out _);
 
-            protected internal override bool TryGetHeaderValues(string name, [NotNullWhen(true)] out IEnumerable<string>? values) => HttpClientTransport.TryGetHeader(_responseMessage.Headers, _responseContent, name, out values);
+            protected internal override IEnumerable<HttpHeader> EnumerateHeaders()
+            {
+                foreach (KeyValuePair<string, string> header in _pipelineResponse.Headers)
+                {
+                    yield return new HttpHeader(header.Key, header.Value);
+                }
+            }
 
-            protected internal override bool ContainsHeader(string name) => HttpClientTransport.ContainsHeader(_responseMessage.Headers, _responseContent, name);
+            protected internal override bool TryGetHeader(string name, [NotNullWhen(true)] out string? value)
+                => _pipelineResponse.Headers.TryGetValue(name, out value);
 
-            protected internal override IEnumerable<HttpHeader> EnumerateHeaders() => GetHeaders(_responseMessage.Headers, _responseContent);
+            protected internal override bool TryGetHeaderValues(string name, [NotNullWhen(true)] out IEnumerable<string>? values)
+                => _pipelineResponse.Headers.TryGetValues(name, out values);
 
-            public override void Dispose()
+            private static void ResetContentStreamPosition(PipelineResponse response)
             {
-                _responseMessage?.Dispose();
-                DisposeStreamIfNotBuffered(ref _contentStream);
+                if (response.ContentStream is MemoryStream &&
```

Comment:
Is the `MemoryStream` type necessary here, or are we including it just to avoid any behavioral changes?

Reply:
It's required because of the current behavior of [HttpMessage.ExtractResponseContent](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/HttpMessage.cs#L190) -- it replaces ContentStream with a stream that throws if you call `CanSeek` on it.  Checking MemoryStream limits the check to cases where either the transport response type did the buffering or a unit test replaced the ContentStream with something custom.